starter-core/src/main/java/io/micronaut/starter/feature/function/template/http/httpFunctionKotlinController.rocker.raw

```diff
@@ -0,0 +1,45 @@
+@import io.micronaut.starter.application.Project
+
+@args (
+Project project
+)
+
+@if (project.getPackageName() != null) {
+package @project.getPackageName()
+}
+
+
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.micronaut.http.annotation.Post
+import io.micronaut.http.annotation.Body
+import io.micronaut.http.HttpStatus
+import io.micronaut.http.annotation.Produces;
+import io.micronaut.http.MediaType;
+
+@@Controller
+class @project.getClassName()Controller {
+
+    @@Produces(MediaType.TEXT_PLAIN)
+    @@Get
+    fun index(): String {
+        return "Example Response"
+    }
+
+    @@Post
+    fun post(@@Body inputMessage: SampleInputMessage?): SampleReturnMessage?{
+        val retMessage = SampleReturnMessage()
+
+        retMessage.returnMessage = "Hello "+inputMessage?.name+", thank you for sending the message"
+        return retMessage
+    }
+}
+
+class SampleInputMessage{
```

Comment:
This should be an [`@Introspected`](https://docs.micronaut.io/latest/guide/index.html#introspection) [Kotlin Data class instead](https://kotlinlang.org/docs/reference/data-classes.html)